docs/core/diagnostics/tutorial_memory_leak.md

```diff
@@ -0,0 +1,243 @@
+
+```dotnetcli
+dotnet-dump collect -p 4807
+```
+
+The result is a core dump located in the same folder.
+
+```console
+Writing minidump with heap to ./core_20190430_185145
+Complete
+```
+
+### Restart the failed process
+
+Once the dump is collected, you should have sufficient information to diagnose the failed process. If the failed process is running on a production server, now is the ideal time for short-term remediation by restarting the process.
+
+In this tutorial, we're now done with the [Sample debug target](sample-debug-target.md). It can now be terminated. Navigate to the terminal that started the server and press `Control-C`.
+
+### Analyze the core dump
+
+Now that you have a core dump generated, use the [dotnet-dump)](dotnet-dump.md) tool to analyze the dump:
+
+```dotnetcli
+dotnet-dump analyze core_20190430_185145
+```
+
+Where `core_20190430_185145` is the name of the core dump you want to analyze.
+
+> [!NOTE]
+> If you see an error complaining that *libdl.so* cannot be found, you may have to install the *libc6-dev* package. For more information, see [Prerequisites for .NET Core on Linux](../linux-prerequisites.md).
+
+You'll be presented with a prompt where you can enter SOS commands. Commonly, the first thing you want to look at is the overall state of the managed heap:
+
+```console
+> dumpheap -stat
+
+Statistics:
+              MT    Count    TotalSize Class Name
+...
+00007f6c1eeefba8      576        59904 System.Reflection.RuntimeMethodInfo
+00007f6c1dc021c8     1749        95696 System.SByte[]
+00000000008c9db0     3847       116080      Free
+00007f6c1e784a18      175       128640 System.Char[]
+00007f6c1dbf5510      217       133504 System.Object[]
+00007f6c1dc014c0      467       416464 System.Byte[]
+00007f6c21625038        6      4063376 testwebapi.Controllers.Customer[]
+00007f6c20a67498   200000      4800000 testwebapi.Controllers.Customer
+00007f6c1dc00f90   206770     19494060 System.String
+Total 428516 objects
+```
+
+Here you can see that most of the objects are either `String` or `Customer` objects.
+
+You can use the `dumpheap` command again with the method table (MT) to get a list of all the `String` instances.
+
+```console
+> dumpheap -mt 00007faddaa50f90
+
+         Address               MT     Size
+...
+00007f6ad09421f8 00007faddaa50f90       94
+...
+00007f6ad0965b20 00007f6c1dc00f90       80
+00007f6ad0965c10 00007f6c1dc00f90       80
+00007f6ad0965d00 00007f6c1dc00f90       80
+00007f6ad0965df0 00007f6c1dc00f90       80
+00007f6ad0965ee0 00007f6c1dc00f90       80
+
+Statistics:
+              MT    Count    TotalSize Class Name
+00007f6c1dc00f90   206770     19494060 System.String
+Total 206770 objects
+```
+
+You can now use the `gcroot` command on a `System.String` instance to see how and why the object is rooted. Be patient for this 30-MB heap this command takes several minutes:
```

Comment:
```suggestion
You can now use the `gcroot` command on a `System.String` instance to see how and why the object is rooted. Be patient because this command takes several minutes with a 30-MB heap:
```